### ISSUE TITLE
Create and Share a working DTS file and Manjaro Arm OS image for the SOQuartz CM4 module on the DeskPi Super6C

### ISSUE THREAD

**Issue by tcdude** [adamfowleruk/deskpi-super6c#1]
Would it be possible to make the DTS you're working on available in this repo? I would love to help out testing and potentially contributing to this effort (I just got my cluster board and soquartz modules today...)

**Comment by adamfowleruk** [comment]
@tcdude Yes, just testing today. Don't want to accidentally blow anyone's kit up! Will upload later today hopefully and let you know.

**Comment by adamfowleruk** [comment]
@tcdude Just added the files. Documented here: https://github.com/adamfowleruk/deskpi-super6c/blob/main/soquartz/image-build-manjaro.md

**Comment by SecKatie** [comment]
I know why it isn't picking up your dtb file. The dtb is set by the bootloader so we would need a custom uboot with the dtb set in this file: https://gitlab.manjaro.org/manjaro-arm/packages/core/uboot-soquartz-cm4/-/blob/main/uboot-soquartz-cm4.install#L12

It should be as simple as modifying the path to point to the new dtb file.

Right now the kernel is not loading your dtb for this and I believe one other reason. We also have to add the new dtb to the Makefile in the dtb directory.

**Comment by SecKatie** [comment]
@adamfowleruk I have created a repo for the custom uboot: https://github.com/JoshuaMulliken/uboot-soquartz-super6c

I also created a repo to hold a version of the upstream manjaro linux package that includes patches to add your dts and also the required makefile changes: https://github.com/JoshuaMulliken/linux-manjaro-pkg-soquartz-super6c

**Comment by SecKatie** [comment]
So I was able to build an image and it does not boot and since the only change is the dtb I think something is not quite right. @adamfowleruk do you have any information that I can use to potentially fix this? How did you create the original version?

**Comment by SecKatie** [comment]
I got it to work!! I had to manually flash the uboot to the sdcard with:

```
tar -xvf uboot-soquartz-{version}.pkg.tar.zst
dd if=uboot-soquartz-{version}.pkg/boot/idbloader.img of=/dev/sdcard seek=64 conv=notrunc,fsync
dd if=uboot-soquartz-{version}.pkg/boot/u-boot.itb of=/dev/sdcard seek=16384 conv=notrunc,fsync
```

The usb ports are still not working though. Here is the dmesg output that shows a couple of failures including bluetooth: https://gist.github.com/JoshuaMulliken/77dca5eaccb1241a096432f1b0a6d6ce

**Comment by SecKatie** [comment]
So I have a script that will perform the entire build. It is very step by step so hopefully it helps everyone to understand where the current issues are.

Issues:
- [ ] USB ports do not work
- [ ] Occasional issues with starting after a restart

https://github.com/JoshuaMulliken/manjaro-for-soquartz-super6c

**Comment by adamfowleruk** [comment]
@SecKatie Hey! Thanks for this. I'm finally getting around to looking at this again now I have the need for a K8S cluster. I've got 4 Rpi CM4s to play with, as well as the SoQuartz module, and a CM4IO board for debugging. I plan on looking at this today. I suspect the USB issue is just the fact the USB hardware (and thus likely the DTS file) is incorrect for the DeskPi board. I'll try debugging this as best I can. So long as I get something that boots I should be able to figure it out. Apologies for the delay in responding.
